api/proto/dop/projectpipeline/projectpipeline.proto

```diff
@@ -46,6 +46,49 @@ service ProjectPipelineService {
       path : "/api/project-pipeline/actions/get-pipeline-yml-list",
     };
   }
+
+  rpc CreateNamePreCheck(CreateProjectPipelineNamePreCheckRequest)
+          returns(CreateProjectPipelineNamePreCheckResponse) {
+    option(google.api.http) = {
+      get : "/api/project-pipeline/actions/name-pre-check",
+    };
+    option(erda.common.openapi) = {
+      path : "/api/project-pipeline/actions/name-pre-check",
+    };
+  }
+
+  rpc CreateSourcePreCheck(CreateProjectPipelineSourcePreCheckRequest)
+          returns(CreateProjectPipelineSourcePreCheckResponse) {
+    option(google.api.http) = {
+      get : "/api/project-pipeline/actions/source-pre-check",
+    };
+    option(erda.common.openapi) = {
+      path : "/api/project-pipeline/actions/source-pre-check",
+    };
+  }
```

Comment:
as a rpc to provide, I think you can combine these two apis to one API like:
`POST /api/project-pipeline/actions/check-create-param`

and the body:
```json
{
  "fields": [
    {
      "field": "name",
      "value": "my name",
    },
    {
      "field": "source",
      "value": "my source",
    }
  ]
}
```